### ISSUE TITLE
Interceptor crashes on Motorola devices

### ISSUE THREAD

**Issue by MuntashirAkon** [MuntashirAkon/AppManager#880]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [ ] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

The interceptor crashes on some devices due to an unknown and apparently weird behaviour. A lot of users have sent me the crash reports via E-Mail, but nobody seems to be willing to help me figure out the issue. I require two information:

1. How to reproduce this crash, that is, what tasks have you performed that caused this crashed?
2. The **framework.jar** file located at `/system/framework/framework.jar` (accessible without root).

### Logs

Example crashlog from the latest version of App Manager:

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{io.github.muntashirakon.AppManager/io.github.muntashirakon.AppManager.intercept.ActivityInterceptor}: java.lang.NumberFormatException: For input string: "83000000"
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3528)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3703)
at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2152)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:250)
at android.app.ActivityThread.main(ActivityThread.java:7886)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:970)
Caused by: java.lang.NumberFormatException: For input string: "83000000"
at java.lang.Integer.parseInt(Integer.java:618)
at java.lang.Integer.valueOf(Integer.java:775)
at java.lang.Integer.decode(Integer.java:1232)
at android.content.Intent.parseUri(Intent.java:7104)
at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.cloneIntent(ActivityInterceptor.java:1036)
at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.storeOriginalIntent(ActivityInterceptor.java:383)
at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.init(ActivityInterceptor.java:370)
at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.onAuthenticated(ActivityInterceptor.java:359)
at io.github.muntashirakon.AppManager.BaseActivity.onCreate(BaseActivity.java:65)
at android.app.Activity.performCreate(Activity.java:8108)
at android.app.Activity.performCreate(Activity.java:8092)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3501)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3703)
at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2152)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:250)
at android.app.ActivityThread.main(ActivityThread.java:7886)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:970)
```

### Device info

App version: 3.0.2
App version code: 413
Android build version: dfc01
Android release version: 11
Android SDK version: 30
Android build ID: RRV31.Q3-15-3-2
Device brand: motorola
Device manufacturer: motorola
Device name: kievv
Device model: motorola one 5G UW ace
Device product name: kievv_vzw
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en
In-App Language: en
Mode: adb_wifi
Inferred Mode: ADB

**Comment by MuntashirAkon** [comment]
Similar crash reports from the earlier versions:

<details>
<summary>Crash reports</summary>

```


java.lang.NumberFormatException: For input string: "83000001"
    at java.lang.Integer.parseInt(Integer.java:747)
    at java.lang.Integer.valueOf(Integer.java:1052)
    at java.lang.Integer.decode(Integer.java:1532)
    at android.content.Intent.parseUriInternal(Intent.java:7376)
    at android.content.Intent.parseUri(Intent.java:7285)
    at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.cloneIntent(ActivityInterceptor.java:1036)
    at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.storeOriginalIntent(ActivityInterceptor.java:383)
    at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.init(ActivityInterceptor.java:370)
    at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.onAuthenticated(ActivityInterceptor.java:359)
    at
io.github.muntashirakon.AppManager.BaseActivity.lambda$onCreate$2$io-github-muntashirakon-AppManager-BaseActivity(BaseActivity.java:106)
    at
io.github.muntashirakon.AppManager.BaseActivity$$ExternalSyntheticLambda2.onChanged(Unknown
Source:6)
    at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
    at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
    at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
    at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
    at androidx.lifecycle.LiveData$1.run(LiveData.java:93)
    at android.os.Handler.handleCallback(Handler.java:978)
    at android.os.Handler.dispatchMessage(Handler.java:104)
    at android.os.Looper.loopOnce(Looper.java:238)
    at android.os.Looper.loop(Looper.java:357)
    at android.app.ActivityThread.main(ActivityThread.java:8090)
    at java.lang.reflect.Method.invoke(Native Method)
    at
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1026)

Device Info:
App version: 3.0.1
App version code: 412
Android build version: b50654-d6017a
Android release version: 12
Android SDK version: 31
Android build ID: S2RES32.29-16-1-2
Device brand: motorola
Device manufacturer: motorola
Device name: denver
Device model: moto g stylus 5G
Device product name: denver_global
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-US
In-App Language: auto
Mode: auto
Inferred Mode: no_root
```

```


java.lang.RuntimeException: Unable to start activity ComponentInfo{io.github.muntashirakon.AppManager/io.github.muntashirakon.AppManager.intercept.ActivityInterceptor}: java.lang.NumberFormatException: For input string: "93080003"
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3528)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3703)
    at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
    at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2152)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loop(Looper.java:250)
    at android.app.ActivityThread.main(ActivityThread.java:7886)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:970)
 Caused by: java.lang.NumberFormatException: For input string: "93080003"
   at java.lang.Integer.parseInt(Integer.java:618)
   at java.lang.Integer.valueOf(Integer.java:775)
   at java.lang.Integer.decode(Integer.java:1232)
   at android.content.Intent.parseUri(Intent.java:7104)
   at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.cloneIntent(ActivityInterceptor.java:1036)
   at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.storeOriginalIntent(ActivityInterceptor.java:383)
   at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.init(ActivityInterceptor.java:370)
   at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.onAuthenticated(ActivityInterceptor.java:359)
   at io.github.muntashirakon.AppManager.BaseActivity.onCreate(BaseActivity.java:65)
   at android.app.Activity.performCreate(Activity.java:8108)
   at android.app.Activity.performCreate(Activity.java:8092)
   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3501)
   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3703)
   at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
   at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
   at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2152)
   at android.os.Handler.dispatchMessage(Handler.java:106)
   at android.os.Looper.loop(Looper.java:250)
   at android.app.ActivityThread.main(ActivityThread.java:7886)
   at java.lang.reflect.Method.invoke(Native Method)
   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:970)

Device Info:
App version: 3.0.1
App version code: 412
Android build version: ac703
Android release version: 11
Android SDK version: 30
Android build ID: RRQS31.Q3-68-149-2
Device brand: motorola
Device manufacturer: motorola
Device name: tonga
Device model: moto g power (2022)
Device product name: tonga_g
Device hardware name: mt6765
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-GB
In-App Language: auto
Mode: auto
Inferred Mode: no_root
```

```

java.lang.NumberFormatException: For input string: "83000000"
at java.lang.Integer.parseInt(Integer.java:618)
at java.lang.Integer.valueOf(Integer.java:775)
at java.lang.Integer.decode(Integer.java:1232)
at android.content.Intent.parseUri(Intent.java:7104)
at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.cloneIntent(ActivityInterceptor.java:1036)
at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.storeOriginalIntent(ActivityInterceptor.java:383)
at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.init(ActivityInterceptor.java:370)
at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.onAuthenticated(ActivityInterceptor.java:359)
at io.github.muntashirakon.AppManager.BaseActivity.lambda$onCreate$2$io-github-muntashirakon-AppManager-BaseActivity(BaseActivity.java:106)
at io.github.muntashirakon.AppManager.BaseActivity$$ExternalSyntheticLambda2.onChanged(Unknown Source:6)
at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
at androidx.lifecycle.LiveData$1.run(LiveData.java:93)
at android.os.Handler.handleCallback(Handler.java:938)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:250)
at android.app.ActivityThread.main(ActivityThread.java:7886)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:970)

Device Info:
App version: 3.0.0
App version code: 410
Android build version: cd0ba
Android release version: 11
Android SDK version: 30
Android build ID: RRV31.Q3-15-3
Device brand: motorola
Device manufacturer: motorola
Device name: kievv
Device model: motorola one 5G UW ace
Device product name: kievv_vzw
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en
In-App Language: en
Mode: adb_tcp
```

```


java.lang.NumberFormatException: For input string: "83000003"
    at java.lang.Integer.parseInt(Integer.java:618)
    at java.lang.Integer.valueOf(Integer.java:775)
    at java.lang.Integer.decode(Integer.java:1232)
    at android.content.Intent.parseUri(Intent.java:7104)
    at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.cloneIntent(ActivityInterceptor.java:1036)
    at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.storeOriginalIntent(ActivityInterceptor.java:383)
    at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.init(ActivityInterceptor.java:370)
    at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.onAuthenticated(ActivityInterceptor.java:359)
    at
io.github.muntashirakon.AppManager.BaseActivity.lambda$onCreate$2$io-github-muntashirakon-AppManager-BaseActivity(BaseActivity.java:106)
    at
io.github.muntashirakon.AppManager.BaseActivity$$ExternalSyntheticLambda2.onChanged(Unknown
Source:6)
    at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
    at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
    at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
    at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
    at androidx.lifecycle.LiveData$1.run(LiveData.java:93)
    at android.os.Handler.handleCallback(Handler.java:938)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loop(Looper.java:250)
    at android.app.ActivityThread.main(ActivityThread.java:7886)
    at java.lang.reflect.Method.invoke(Native Method)
    at
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:970)

Device Info:
App version: 3.0.0
App version code: 410
Android build version: d67faa
Android release version: 11
Android SDK version: 30
Android build ID: RRHS31.Q3-46-110-10
Device brand: motorola
Device manufacturer: motorola
Device name: ellis
Device model: moto g pure
Device product name: ellis_trac
Device hardware name: mt6762
ABIs: [armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): []
System language: en-US
In-App Language: auto
Mode: auto
```

```


java.lang.RuntimeException: Unable to start activity
ComponentInfo{io.github.muntashirakon.AppManager/io.github.muntashirakon.AppManager.intercept.ActivityInterceptor}:
java.lang.NumberFormatException: For input string: "83000001"
    at
android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3528)
    at
android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3703)
    at
android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
    at
android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
    at
android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2152)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loop(Looper.java:250)
    at android.app.ActivityThread.main(ActivityThread.java:7886)
    at java.lang.reflect.Method.invoke(Native Method)
    at
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:970)
Caused by: java.lang.NumberFormatException: For input string: "83000001"
   at java.lang.Integer.parseInt(Integer.java:618)
   at java.lang.Integer.valueOf(Integer.java:775)
   at java.lang.Integer.decode(Integer.java:1232)
   at android.content.Intent.parseUri(Intent.java:7104)
   at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.cloneIntent(ActivityInterceptor.java:885)
   at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.storeOriginalIntent(ActivityInterceptor.java:354)
   at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.init(ActivityInterceptor.java:343)
   at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.onAuthenticated(ActivityInterceptor.java:332)
   at
io.github.muntashirakon.AppManager.BaseActivity.onCreate(BaseActivity.java:46)
   at android.app.Activity.performCreate(Activity.java:8108)
   at android.app.Activity.performCreate(Activity.java:8092)
   at
android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
   at
android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3501)
   at
android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3703)
   at
android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
   at
android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
   at
android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2152)
   at android.os.Handler.dispatchMessage(Handler.java:106)
   at android.os.Looper.loop(Looper.java:250)
   at android.app.ActivityThread.main(ActivityThread.java:7886)
   at java.lang.reflect.Method.invoke(Native Method)
   at
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:970)

Device Info:
App version: 2.6.5.1
App version code: 397
Android build version: c170ed
Android release version: 11
Android SDK version: 30
Android build ID: RRES31.Q3-60-47-2-2-2
Device brand: motorola
Device manufacturer: motorola
Device name: denver
Device model: moto g stylus 5G
Device product name: denver_global
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-US
In-App Language: auto
Mode: auto
```

```
java.lang.RuntimeException: Unable to start activity
ComponentInfo{io.github.muntashirakon.AppManager/io.github.muntashirakon.AppManager.intercept.ActivityInterceptor}:
java.lang.NumberFormatException: For input string: "9b080001"
    at
android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3528)
    at
android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3703)
    at
android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
    at
android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
    at
android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2152)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loop(Looper.java:250)
    at android.app.ActivityThread.main(ActivityThread.java:7886)
    at java.lang.reflect.Method.invoke(Native Method)
    at
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:970)
Caused by: java.lang.NumberFormatException: For input string: "9b080001"
   at java.lang.Integer.parseInt(Integer.java:618)
   at java.lang.Integer.valueOf(Integer.java:775)
   at java.lang.Integer.decode(Integer.java:1232)
   at android.content.Intent.parseUri(Intent.java:7104)
   at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.cloneIntent(ActivityInterceptor.java:977)
   at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.storeOriginalIntent(ActivityInterceptor.java:361)
   at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.init(ActivityInterceptor.java:348)
   at
io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.onAuthenticated(ActivityInterceptor.java:337)
   at
io.github.muntashirakon.AppManager.BaseActivity.onCreate(BaseActivity.java:65)
   at android.app.Activity.performCreate(Activity.java:8108)
   at android.app.Activity.performCreate(Activity.java:8092)
   at
android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
   at
android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3501)
   at
android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3703)
   at
android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
   at
android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
   at
android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2152)
   at android.os.Handler.dispatchMessage(Handler.java:106)
   at android.os.Looper.loop(Looper.java:250)
   at android.app.ActivityThread.main(ActivityThread.java:7886)
   at java.lang.reflect.Method.invoke(Native Method)
   at
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:970)

Device Info:
App version: 3.0.0-alpha01
App version code: 400
Android build version: 1a1177
Android release version: 11
Android SDK version: 30
Android build ID: RRMS31.Q3-23-77-1-2
Device brand: motorola
Device manufacturer: motorola
Device name: berlna
Device model: motorola edge (2021)
Device product name: berlna_global
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-US
In-App Language: auto
Mode: auto
```
<details>

**Comment by MuntashirAkon** [comment]
Ah, all of them seems to be coming from Motorola devices. So, I guess the issue is specific to Motorola devices.

**Comment by gnuhead-chieb** [comment]
https://ufile.io/9ujjoe50
Just mirror,same:https://anonfiles.com/m5a4e2Ady4/framework_jar
Here is extracted framework from Motorola XT2131-3(Denver) stock ROM RRES31.Q3-60-47-2-2-2

I extracted from downloaded ROM. I dont have motorola phone,so I'm not able to test for this bug.

**Comment by gnuhead-chieb** [comment]
If you need files anything else,I can send them.Here is the lists of files
[product_a.img.txt](https://github.com/MuntashirAkon/AppManager/files/9659070/product_a.img.txt)
[system_a.img.txt](https://github.com/MuntashirAkon/AppManager/files/9659071/system_a.img.txt)
[system_b.img.txt](https://github.com/MuntashirAkon/AppManager/files/9659072/system_b.img.txt)
[system_ext_a.img.txt](https://github.com/MuntashirAkon/AppManager/files/9659075/system_ext_a.img.txt)
[vendor_a.img.txt](https://github.com/MuntashirAkon/AppManager/files/9659077/vendor_a.img.txt)

**Comment by MuntashirAkon** [comment]
Figured out the issue. The number `0x83000000` is `2197815296` which is beyond what an `Integer` can support. What is interesting is that how it is able to produce this value. If what you've shared is indeed the source, this value is not possible at all. Clearly, I need to know what caused it.

**Comment by shuvashish76** [comment]
> 1. How to reproduce this crash, that is, what tasks have you performed that caused this crashed?

Interceptor crashed while testing an OCR app.

Steps to reproduce:
1. Install [OCR](https://www.f-droid.org/packages/io.github.subhamtyagi.ocr/) app
2. Download English OCR training data.
3. Click & choose AM Interceptor
AM crashes

<details>
  <summary>Crash logs:</summary>

```
java.lang.NumberFormatException: For input string: "83000000"
    at java.lang.Integer.parseInt(Integer.java:747)
    at java.lang.Integer.valueOf(Integer.java:1052)
    at java.lang.Integer.decode(Integer.java:1532)
    at android.content.Intent.parseUriInternal(Intent.java:7376)
    at android.content.Intent.parseUri(Intent.java:7285)
    at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.cloneIntent(ActivityInterceptor.java:1036)
    at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.storeOriginalIntent(ActivityInterceptor.java:383)
    at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.init(ActivityInterceptor.java:370)
    at io.github.muntashirakon.AppManager.intercept.ActivityInterceptor.onAuthenticated(ActivityInterceptor.java:359)
    at io.github.muntashirakon.AppManager.BaseActivity.lambda$onCreate$2$io-github-muntashirakon-AppManager-BaseActivity(BaseActivity.java:106)
    at io.github.muntashirakon.AppManager.BaseActivity$$ExternalSyntheticLambda2.onChanged(Unknown Source:6)
    at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
    at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
    at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
    at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
    at androidx.lifecycle.LiveData$1.run(LiveData.java:93)
    at android.os.Handler.handleCallback(Handler.java:978)
    at android.os.Handler.dispatchMessage(Handler.java:104)
    at android.os.Looper.loopOnce(Looper.java:238)
    at android.os.Looper.loop(Looper.java:357)
    at android.app.ActivityThread.main(ActivityThread.java:8090)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1026)

```
Device Info:
App version: 3.0.4
App version code: 416
Android build version: e0e06-7bba9c
Android release version: 12
Android SDK version: 31
Android build ID: S2RUBS32.51-15-9-1
Device brand: motorola
Device manufacturer: motorola
Device name: corfur
Device model: moto g71 5G
Device product name: corfur_g
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-IN
In-App Language: auto
Mode: auto
Inferred Mode: no_root

https://user-images.githubusercontent.com/48226312/212416649-6639cd66-e350-4a8f-bf65-db54b68fb6eb.mp4
</details>

> 2. The framework.jar file located at /system/framework/framework.jar (accessible without root).

File sent.

**Comment by MuntashirAkon** [comment]
> java.lang.NumberFormatException: For input string: "83000000"

The flag was supposed to be `0x3000000`, not `0x83000000`.

**Comment by MuntashirAkon** [comment]
Try the latest debug build. Whether it crashes or not, attach the am.log file located at `/sdcard/Android/data/io.github.muntashirakon.AppManager.debug/cache/`.

**Comment by shuvashish76** [comment]
V3.1.0-alpha02-DEBUG (415)

[am.log](https://github.com/MuntashirAkon/AppManager/files/10417226/am.log)

**Comment by MuntashirAkon** [comment]
Ah, `Integer.decode()` expects a _signed_ hexadecimal integer but `Integer.toHexString` returns an _unsigned_ hexadecimal integer which is causing the issue. This issue would be quite difficult to fix in my side, but this should be considered a bug in the AOSP. I shall file a report regarding the issue.

**Comment by MuntashirAkon** [comment]
Fixed in b35634a5fa0442d35afbf1911fff3216d8b44c88
